File: server/Topics.Data/Entities/Topic.cs
namespace Topics.Data.Entities;

public class Topic
{
    public int Id { get; set; }
    public int OwnerId { get; set; }
    public string Name { get; set; }
    public string Url { get; set; }
    public DateTime DateCreated { get; set; }

    public User Owner { get; set; }

    /* One to One */
    public TopicImage Image { get; set; }

    public IEnumerable<Post> Posts { get; set; }
    public IEnumerable<TopicUser> Members { get; set; }
}

File: client/core/models/api/topic.ts
import { Post } from './post';
import { TopicImage } from './upload';
import { TopicUser } from './topic-user';
import { User } from './user';

export interface Topic {
  id: number;
  ownerId: number;
  name: string;
  url: string;
  dateCreated: string;

  owner?: User;

  image?: TopicImage;

  posts?: Post[];
  members?: TopicUser[];
}
